spec/ics-3-connection-semantics/README.md

```diff
@@ -0,0 +1,245 @@
+
+This standards document outlines the abstraction of an IBC _connection_: the state and consensus ruleset necessary to perform IBC packet verification.
+
+## Specification
+
+### Motivation
+
+- Connection = cross-chain light client state.
+- Between two chains `A` and `B`.
+- Permissionless opening / closing / updates.
+
+### Desired Properties
+
+- Permissionless channel opening / channel closing / channel updates.
+
+#### Pre-Establishment
+
+- Guarantees that no packets can be committed on other connections?
+- No required a priori root-of-trust knowledge
+- Only one connection "between" two chains
+
+#### During Handshake
+
+Once a negotiation handshake has *begun* (defined as the first packet being committed):
+
+- Only the appropriate handshake packets can be committed in order
+- No chain can masquerade as one of the handshaking chains (formalize...)
+
+#### Post-Establishment
+
+- Connection provides verification of relayed packets
+- No packets other than committed by consensus of *A* / *B* on connection *C* can be relayed
+
+### Technical Specification
+
+(detailed technical specification: syntax, semantics, sub-protocols, algorithms, data structures, etc)
+
+#### Definitions
+
+- Chain `A` is the source blockchain from which the IBC packet is sent
+- Chain `B` is the destination blockchain on which the IBC packet is received
+- `H_h` is the signed header of chain `A` at height `h` 
+- `C_h` is a subset of the consensus ruleset of chain `A` at height `h` 
+- `V_kh` is the value stored on chain `A` under key `k` at height `h` 
+- `P` is the unbonding period of chain `P`, in units of time
+- `dt(a, b)` is the time difference between events `a` and `b` 
+
+Note that of all these, only `H_h` defines a signature and is thus attributable.
+
+#### Requirements
+
+(reference ICS 2: consensus primitives)
+
+To facilitate an IBC connection, the two blockchains must provide the following proofs:
+
+1. Given a trusted `H_h` and `C_h` and an attributable update message `U_h`,  
+   it is possible to prove `H_h'` where `C_h' == C_h` and `dt(now, H_h) < P`
+2. Given a trusted `H_h` and `C_h` and an attributable change message `X_h`,  
+   it is possible to prove `H_h'` where `C_h' /= C_h` and `dt(now, H_h) < P`
+3. Given a trusted `H_h` and a Merkle proof `M_kvh` it is possible to prove `V_kh`
+
+It is possible to make use of the structure of BFT consensus to construct extremely lightweight and provable messages `U_h'` and `X_h'`. The implementation of these requirements with Tendermint consensus is defined in [Appendix E](appendices.md#appendix-e-tendermint-header-proofs). Another algorithm able to provide equally strong guarantees (such as Casper) is also compatible with IBC but must define its own set of update and change messages.
```

Comment:
Paragraph probably doesn't belong here.

Reply:
All related discussion and more specific requirements moved to ICS 2.